Fixes thread_auto_close execution when disabled.

martinbndr opened the pull request:
This fixes the issue #3290 which caused threads to be auto-closed even if `thread_auto_close` has been disabled.

There was also an issue that closed the thread when the user has responded to mods. The thread should stay open and only auto close when the staff has replied back.

martinbndr committed:
Fixes thread_auto_close execution when disabled.

This fixes the issue #3290 which caused threads to be auto-closed even if `thread_auto_close` has been disabled.

There was also an issue that closed the thread when the user has responded to mods. The thread should stay open and only auto close when the staff has replied back.

lorenzo132 committed:
fix: prevent autoclosing when close has been cancelled.

This solves the thread from autoclosing if the closure has been cancelled earlier in a thread.

sebkuip commented on cogs/utility.py:
```diff
@@ -873,6 +873,20 @@ async def config_remove(self, ctx, *, key: str.lower):
+                for recipient_id, items in tuple(closures.items()):
+                    if items.get("auto_close", False) is True:
+                        self.bot.config["closures"].pop(recipient_id)
+                        await self.config.update()
```
Is it not possible to call config.update() at the end of the for loop? If you have a decent amount of thread pending to be cancelled, you'd get a lot of mongo data transfer that doesn't seem needed.

sebkuip commented on cogs/utility.py:
```diff
@@ -873,6 +873,20 @@ async def config_remove(self, ctx, *, key: str.lower):
+                            await thread.cancel_closure(all=True)
+                        else:
+                            self.bot.config["closures"].pop(recipient_id)
+                            await self.config.update()
```
Same as above.

sebkuip reviewed (changes requested):
There's some extra mongo calls that I think could be removed to save bandwidth limits. Otherwise seems okay

lorenzo132 committed:
fix: AttributeError / lower mongo calls.

I had added a small bugfix aswell for pagination when an invalid config var was given.

This happened to occur upon removing the `thread_auto_close` config.

sebkuip reviewed (approved):
LGTM

lorenzo132 committed:
Merge branch 'development' into users/martinbndr/auto-close
